### PR TITLE
Duration

### PR DESCRIPTION
**Pull request by TheLillyleaf**
Added `duration` to tidily as an input alternative. 

* Will take the format `hh*:mm `
* will format the input as `123:45 h`
* Will allow any length of h but only up to 59 m

Also fixed some spelling error on multiple files

### CODE REVIEW

File: Handler/duration.spec.ts [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+		expect(result).toMatchObject({ value: "12:23 h", selection: { start: 7, end: 7 } })
+	})
+	it("Add 60 minutes", () => {
+		const result = Action.apply(handler, { value: "12:6", selection: { start: 4, end: 4 } }, { key: "0" })
+		expect(result).toMatchObject({ value: "12:6 h", selection: { start: 6, end: 6 } })
+	})
+	it("Key event backspace", () => {
+		let result = { value: "12:34", selection: { start: 5, end: 5 } }
+		result = Action.apply(handler, result, { key: "Backspace" })
+		expect(result).toMatchObject({ value: "12:3 h", selection: { start: 6, end: 6 } })
```
Comment (EliasEriksson) [inline Handler/duration.spec.ts]: is this 12:03? if it is we should use 0 padding using `padStart(...)`. If it is 12:30 we should use `padEnd(...)`

File: Handler/duration.spec.ts [hunk 2]
```diff
@@ -0,0 +1,37 @@
...
+	})
+	it("Add too many minutes", () => {
+		const result = Action.apply(handler, { value: "12:23", selection: { start: 5, end: 5 } }, { key: "3" })
+		expect(result).toMatchObject({ value: "12:23 h", selection: { start: 7, end: 7 } })
+	})
+	it("Add 60 minutes", () => {
+		const result = Action.apply(handler, { value: "12:6", selection: { start: 4, end: 4 } }, { key: "0" })
+		expect(result).toMatchObject({ value: "12:6 h", selection: { start: 6, end: 6 } })
+	})
+	it("Key event backspace", () => {
```
Comment (EliasEriksson) [inline Handler/duration.spec.ts:27]: what happens if the end result is an empty string when backspace is used?

File: Handler/duration.ts [hunk 3]
```diff
@@ -0,0 +1,37 @@
...
+	}
+	fromString(value: string): { hours: number; minutes: number } | undefined {
+		const splitted = typeof value == "string" && value.split(":", 2).map(Number.parseInt)
+		return splitted ? { hours: splitted[0], minutes: splitted[1] } : undefined
+	}
+	format(unformatted: StateEditor): Readonly<State> & Settings {
+		let result = unformatted
+		if (result.value.length > 0)
+			result = result.append(" h")
+		return { ...result, type: "tel", pattern: /^\d*:{0,1}[0-5]{0,1}[0-9]{0,1}(\sh{0,1}){0,1}$/ }
```
Comment (EliasEriksson) [inline Handler/duration.ts:26]: what does type tel do here? is it to get a number input without the arrows?

**Review by EliasEriksson — CHANGES_REQUESTED** [review]

**Comment by fredrikbryntesson** [comment]
This commit will have he name "Start of duration" when it gets merged. Will that be sufficient for that it actually is or does it need to be updated?

**Review by EliasEriksson — APPROVED** [review]

**Pull request merged**